modules/Xsens_module/include/XsensRetargeting.hpp

```diff
@@ -0,0 +1,104 @@
+#ifndef XSENSRETARGETING_H
+#define XSENSRETARGETING_H
+
+// std
+#include <cmath>
+#include <memory>
+
+// YARP
+#include <yarp/os/Bottle.h>
+
+// iCub-ctrl
+#include <iCub/ctrl/minJerkCtrl.h>
+#include <yarp/dev/IFrameTransform.h>
+#include <yarp/dev/IJoypadController.h>
+#include <yarp/dev/PolyDriver.h>
+#include <yarp/os/Bottle.h>
+#include <yarp/os/BufferedPort.h>
+#include <yarp/os/RFModule.h>
+#include <yarp/os/RpcClient.h>
+#include <yarp/sig/Vector.h>
+
+#include <chrono>
+#include <yarp/os/Clock.h>
+// iDynTree
+#include <iDynTree/Core/Transform.h>
+//#include <RetargetingController.hpp>
+
+class mapJoints
+{
+public:
+    std::string name;
+    int index;
+};
+
+class XsensRetargeting : public yarp::os::RFModule
+{
+private:
+    /** Minimum jerk trajectory smoother for the desired whole body joints */
+    std::unique_ptr<iCub::ctrl::minJerkTrajGen> m_WBTrajectorySmoother{nullptr};
+    yarp::sig::Vector m_jointValues, m_smoothedJointValues;
+    std::vector<std::string>
+        m_humanJointsListName; // the order of joints list arrived from human state provider is
+                               // different from the one we want to send to the controller
+
+    /** Port used to retrieve the human whole body joint pose. */
+    yarp::os::BufferedPort<yarp::os::Bottle> m_wholeBodyHumanJointsPort;
+
+    /** Port used to retrieve the human whole body joint pose. */
+    yarp::os::BufferedPort<yarp::sig::Vector> m_wholeBodyHumanSmoothedJointsPort;
+
+    double m_dT; /**< Module period. */
+    bool m_useXsens; /**< True if the Xsens is used in the retargeting */
+
+    std::vector<std::string>
+        m_robotJointsListNames; /**< Vector containing the name of the controlled joints.*/
+    size_t m_actuatedDOFs; /**< Number of the actuated DoF */
+
+    std::vector<unsigned> m_humanToRobotMap;
+
+    //   std::chrono::milliseconds m_tick,m_tock;
```

Comment:
if  `m_tick` and `m_tock` are not used you can remove them

Reply:
done.